File: backend/Domain/Entities/GuestHouse.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Domain.Common;

namespace Domain.Entities
{
    public class GuestHouse : BaseEntity
    {
        [ForeignKey(nameof(ResidentialProperty))]
        public Guid ResidentialPropertyId { get; set; }
        [Range(1, 5)]
        public int StarRating { get; set; }
        public bool RestaurantOnSite { get; set; }

        public ResidentialProperty ResidentialProperty { get; set; } = null!;
    }
}

File: mobile/lib/src/features/dashboard/domain/entities/guest_house.dart
import 'package:equatable/equatable.dart';

class GuestHouse extends Equatable {
  final int starRating;
  final bool restaurantOnSite;

  const GuestHouse({
    required this.starRating,
    required this.restaurantOnSite,
  });

  @override
  List<Object?> get props => [starRating, restaurantOnSite];
}
